pandas/tests/extension/base/methods.py

```diff
@@ -394,7 +394,10 @@ def test_hash_pandas_object_works(self, data, as_frame):
 
     def test_searchsorted(self, data_for_sorting, as_series):
         b, c, a = data_for_sorting
-        arr = type(data_for_sorting)._from_sequence([a, b, c])
+        # Pass dtype so sorting order is not lost for Categorical
+        arr = type(data_for_sorting)._from_sequence(
+            [a, b, c], dtype=data_for_sorting.dtype
+        )
```

Comment:
not sure if we should be changing base extensions tests. @jorisvandenbossche?

Reply:
I think that's fine to change it here (all EAs should be able to handle a passed dtype to `_from_sequence`, so shouldn't influence the test for others)

Now, it can probably also be done in a shorter way, something like `arr = data_for_sorting.take([2, 0, 1])  # to get [a, b, c]`

Reply:
Thanks @jorisvandenbossche, will update with the shortened form.